### PR TITLE
Minor Tasks Rest Api fixes

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

File: src/python/foglamp/core/api/scheduler.py [hunk 1]
```diff
@@ -21,14 +21,17 @@
 _help = """
     -------------------------------------------------------------------------------
     | GET             | /foglamp/schedule/process                                 |
+    | GET             | /foglamp/schedule/process/{scheduled_process_id}          |
 
-    | GET             | /foglamp/schedules                                        |
+    | GET             | /foglamp/schedule                                         |
     | POST            | /foglamp/schedule                                         |
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/api/scheduler.py]: Please do in single line | GET POST            | /foglamp/schedule

File: src/python/foglamp/core/routes.py [hunk 2]
```diff
@@ -32,23 +32,22 @@ def setup(app):
 
     # Scheduler
     # Scheduled_processes - As per doc
-    app.router.add_route('GET', '/foglamp/scheduled/processes', api_scheduler.get_scheduled_processes)
-    app.router.add_route('GET', '/foglamp/scheduled/process/{scheduled_process_name}', api_scheduler.get_scheduled_process)
+    app.router.add_route('GET', '/foglamp/schedule/process', api_scheduler.get_scheduled_processes)
+    app.router.add_route('GET', '/foglamp/schedule/process/{scheduled_process_name}', api_scheduler.get_scheduled_process)
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/routes.py]: What is the significance of this method? Even not documented on g-drive.
```
URL: {{base_uri}}/foglamp/schedule/process/purge
Response: "purge"
```

File: src/python/foglamp/core/api/scheduler.py [hunk 3]
```diff
@@ -21,14 +21,17 @@
 _help = """
     -------------------------------------------------------------------------------
     | GET             | /foglamp/schedule/process                                 |
+    | GET             | /foglamp/schedule/process/{scheduled_process_id}          |
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/api/scheduler.py]: scheduled_process_name or scheduled_process_id?

File: src/python/foglamp/core/routes.py [hunk 4]
```diff
@@ -32,23 +32,22 @@ def setup(app):
...
 
     # Tasks - As per doc
-    app.router.add_route('GET', '/foglamp/tasks', api_scheduler.get_tasks)
-    app.router.add_route('GET', '/foglamp/tasks/latest', api_scheduler.get_tasks_latest)
+    app.router.add_route('GET', '/foglamp/task', api_scheduler.get_tasks)
+    app.router.add_route('GET', '/foglamp/task/latest', api_scheduler.get_tasks_latest)
     app.router.add_route('GET', '/foglamp/task/{task_id}', api_scheduler.get_task)
-    # TODO: Find out why not DELETE/PUT a task? Cancel flag can be handled in database update.
-    app.router.add_route('POST', '/foglamp/task/{task_id}/cancel', api_scheduler.cancel_task)
+    app.router.add_route('PUT', '/foglamp/task/cancel/{task_id}', api_scheduler.cancel_task)
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/routes.py]: update the url on documentation itself

File: src/python/foglamp/core/routes.py (hunk 2 above)
Comment (amarendra-dianomic) [inline src/python/foglamp/core/routes.py]: Required as we need to verify on server side a valid process name in creating a schedule.

File: src/python/foglamp/core/routes.py (hunk 4 above)
Comment (amarendra-dianomic) [inline src/python/foglamp/core/routes.py]: Will ask Mark for the same.

File: src/python/foglamp/core/api/scheduler.py [hunk 5]
```diff
@@ -490,7 +545,21 @@ def _extract_args(data, curr_value):
...
+                {'id': str(task['id']),
+                     'process_name': task['process_name'],
+                     'state': [t.name for t in list(Task.State)][int(task['state']) - 1],
+                     'start_time': str(task['start_time']),
+                     'end_time': str(task['end_time']),
+                     'exit_code': task['exit_code'],
+                     'reason': task['reason'],
+                     'pid': task['pid']
+                 }
+            )
```
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api/scheduler.py]: done

File: src/python/foglamp/core/api/scheduler.py (hunk 1 above)
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api/scheduler.py]: done

File: src/python/foglamp/core/api/scheduler.py (hunk 3 above)
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api/scheduler.py]: done

File: src/python/foglamp/core/api/scheduler.py [hunk 6]
```diff
@@ -12,7 +12,7 @@
 from foglamp.core.api import scheduler_db_services
 from foglamp.core.scheduler import Schedule, Scheduler, StartUpSchedule, TimedSchedule, IntervalSchedule, ManualSchedule, Task
```
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api/scheduler.py]: done

**Comment by amarendra-dianomic** [comment]
@ashish-ScaleDB re post_schedule, update_schedule, the updated/added schedule is being fetched by get_schedule method, hence schedule_id is being duplicated. Just to remove one key, we will need to do substantial code change, test the same again etc, hence suggest that let it remain as it is.

**Comment by amarendra-dianomic** [comment]
@ashish-ScaleDB all done, pl check.

**Comment by ashish-ScaleDB** [comment]
You just only need to remove this key-pair from posting update schedule api methods. ~{'id': str(updated_schedule_id)~
```
return web.json_response({'schedule': schedule})
```

**Pull request merged**